client/electron/go_plugin.ts

```diff
@@ -36,7 +34,105 @@ export async function invokeGoMethod(
   method: string,
   input: string
 ): Promise<string> {
-  if (!invokeMethodFunc) {
+  console.debug(`[Backend] - calling InvokeMethod "${method}" ...`);
+  const result = await ensureCgo().invokeMethod(method, input);
+  console.debug(`[Backend] - InvokeMethod "${method}" returned`, result);
+  if (result.ErrorJson) {
+    throw new Error(result.ErrorJson);
+  }
+  return result.Output;
+}
+
+/**
+ * Represents a function that will be called from the Go backend.
+ * @param data The data string passed from the Go backend.
+ */
+export type CallbackFunction = (data: string) => void;
+
+/** A unique token for a callback created by `newCallback`. */
+export type CallbackToken = string;
+
+/**
+ * Koffi requires us to register all persistent callbacks; we track the registrations here.
+ * @see https://koffi.dev/callbacks#registered-callbacks
+ */
+const koffiCallbacks = new Map<string, koffi.IKoffiRegisteredCallback>();
```

Comment:
No need to change for this PR since it's already implemented, but there's an alternative design where the callback mapping you do in the Go code is done in the Typescript code instead. That's what Cordova does. This would circumvent the restrictions from koffi, since all you need in that case is a single callback.

Reply:
Actually [Koffi's limit](https://koffi.dev/callbacks#registered-callbacks) is pretty big. We might not event need that much of callbacks in the entire system.

> A maximum of 8192 callbacks can exist at the same time.